storage/src/main/java/org/apache/kafka/server/log/remote/storage/RemoteLogManager.java

```
@@ -1044,6 +1049,9 @@ private void copyLogSegment(UnifiedLog log, LogSegment segment, RemoteLogSegment
             
             try {
                 customMetadata = remoteStorageManagerPlugin.get().copyLogSegmentData(copySegmentStartedRlsm, segmentData);
+            } catch (RetriableRemoteStorageException e) {
+                logger.info("Copy failed with retriable error for segment {}", copySegmentStartedRlsm.remoteLogSegmentId());
+                throw e;
             } catch (RemoteStorageException e) {
                 logger.info("Copy failed, cleaning segment {}", copySegmentStartedRlsm.remoteLogSegmentId());
                 try {
```

In L1058 below, when `deleteRemoteLogSegment` throws `RetriableRemoteStorageException`, we don't catch it. Then, the exception thrown will be the one from `deleteRemoteLogSegment`, not the one from `copyLogSegmentData`. With that, the `RetriableRemoteStorageException` from `deleteRemoteLogSegment` could be caught in [L1009](https://github.com/apache/kafka/pull/21150/files#diff-b5803060ec32fdfea597965205edfaa9949566a097f7017f4800ea4dee86ea9bR1009) above and not update the metrics as expected. Maybe we should catch all exceptions thrown from L1058 because we want to retry later?

---

We also have to handle the RetriableRemoteStorageException in RLMTask L844, so that it won't be logged as WARN log. 

```
} catch (RetriableException | RetriableRemoteStorageException ex) {
                logger.debug("Encountered a retryable error while executing current task for partition {}", topicIdPartition, ex);
            }
```

---

@DL1231 Could you also address the above comment? Thanks!

---

> Maybe we should catch all exceptions thrown from L1058 because we want to retry later?

since RRSE extends RSE, the current patch looks clean to me. (Open to suggestions)

---

> when deleteRemoteLogSegment throws RetriableRemoteStorageException, we don't catch it.

as @kamalcph pointed out, RRSE extends RSE, so when `deleteRemoteLogSegment` throws RRSE, it will be caught.

---

Ah, OK, `RRSE extends RSE` <-- thanks for pointing it out.